### PR TITLE
chore: Add route and compression handling to test-server

### PR DESCRIPTION
**Pull request by metal-messiah**
Add route and compression handling to test-server in preparation for SR.  SR will report compressed data payloads in a non-standard way, by including a query param which notes its compression.  GZIP handling as well as preprocessing were added to the testing server to facilitate this.
---

### Overview
Add route and compression handling to test-server in preparation for SR.  A positive side-effect of this is that the testing-server can now return compressed scripts and pages, which may decrease the total burden on sauce-connection issues.

### Related Issue(s)
NEWRELIC-8807

### Testing
The test server has a new route `/blob` which can handle compressed payloads and expects data shaped as defined in the `/blob` endpoint spec found here --> https://newrelic.atlassian.net/l/cp/CjDxzmo3

### CODE REVIEW

File: src/features/session_replay/aggregate/index.js [hunk 1]
```diff
@@ -153,7 +149,7 @@ export class Aggregate extends AggregateBase {
       if (fullSample) this.mode = MODE.FULL // full mode has precedence over error mode
       else if (errorSample) this.mode = MODE.ERROR
       // If neither are selected, then don't record (early return)
-      return
+      else return
```
Comment (patrickhousley) [inline src/features/session_replay/aggregate/index.js:152]: Same difference

File: tools/testing-server/routes/mock-apis.js [hunk 2]
```diff
@@ -110,7 +110,9 @@ module.exports = fp(async function (fastify, testServer) {
   fastify.get('/js', (request, reply) => {
     reply.type('text/javascript').send('console.log(\'hi\')')
   })
-  fastify.get('/text', (request, reply) => {
+  fastify.get('/text', {
+    compress: false
+  }, (request, reply) => {
```
Comment (patrickhousley) [inline tools/testing-server/routes/mock-apis.js:115]: Should we just add this to all the mock api routes?

File: src/features/session_replay/aggregate/index.js (hunk 1 above)
Comment (metal-messiah) [reply src/features/session_replay/aggregate/index.js:152]: Not exactly. The else is Basically saying if it's "off" then return (ie don't run the rest of the function)

If we always return it'll never run

Comment (metal-messiah) [reply src/features/session_replay/aggregate/index.js:152]: Not exactly. The else is Basically saying if it's "off" then return (ie don't run the rest of the function)

If we always return it'll never run

File: tools/testing-server/routes/mock-apis.js (hunk 2 above)
Comment (metal-messiah) [reply tools/testing-server/routes/mock-apis.js:115]: Probably tbh

File: src/features/session_replay/aggregate/index.js (hunk 1 above)
Comment (patrickhousley) [reply src/features/session_replay/aggregate/index.js:152]: Ah lol, I didn't see the rest of the code.

**Review by patrickhousley — APPROVED** [review]

**Review by bjfield — APPROVED** [review]
Using `rrweb-record.html?loader=experimental` and some console logging, I verified that the agent is sending gzipped content and the blob test endpoint is unzipping the body.

**Pull request merged**